feat!: Add client version to ENR

mralj opened the pull request:
Closes #14124

The client version is added to the ENR. The alternative was to add it to the `Status` message, but I decided on this approach, as the team suggested.  

### Implementation details
The client version is injected into `discV5_service` instead of simply calling `getPackageVersion()` in the service constructor. The reason behind this is simple: it seemed more correct to do it this way in case there is any change in versioning logic in the future. This way, we make sure the caller of the P2P service and the P2P service always have the same version.

### Refactoring
It seemed more appropriate to implement `getPackageVersion` in the `stdlib` since it's now used across more packages. 
Furthermore, I added logic to handle erroneous cases when reading packageVersion instead of simply throwing the error.

The ENR generation is also extracted into the already existing `generate-enr` and follows the logic of ENR generation for the bootstrap nodes. 

### Bootstrap nodes
I deliberately omitted the client version from bootstrap nodes ENR and did not change the bootstrap node's ENR with each version change, as this seems counterproductive and counterintuitive. 

The alternative was for Bootstrap nodes to either have a hardcoded version (e.g., `0` or `0.0.0`) or follow alternative versioning (e.g., `boot-v1`). However, this also seemed like a worse idea because handling boot node versions would require special logic and, if not handled correctly, could cause confusion when displaying this data.

mralj committed:
Added client version to ENR

mralj committed:
Added client version to ENR

mralj committed:
Added client version to ENR

mralj committed:
Added client version to ENR

mralj committed:
Added client version to ENR

mralj committed:
Extracted ENR creation to separate function

mralj committed:
Extracted ENR creation to separate function

mralj committed:
Extracted ENR creation to separate function

mralj committed:
Extracted ENR creation to separate function

mralj committed:
Extracted ENR creation to separate function

PhilWindle commented on yarn-project/stdlib/src/versioning/versioning.ts:
```diff
@@ -145,3 +149,28 @@ export function getVersioningResponseHandler(versions: Partial<ComponentsVersion
+ * If .release-please-manifest.json could not be read or parsed returns empty string
+ */
+export function getPackageVersion(logger?: Logger): string {
+  try {
```
@alexghr We will want to consolidate this with the work we are doing around versioning.

PhilWindle committed:
Merge branch 'master' into feat/enr-client-version

mralj committed:
Merge branch 'AztecProtocol:master' into feat/enr-client-version

alexghr committed:
feat: add auto updater (#14292)

Fix #14238

---------

Co-authored-by: Santiago Palladino <santiago@aztecprotocol.com>

TomAFrench committed:
fix: make `msgpackr` a full dependency of `bb.js` (#14322)

This PR fixes the issue in bb.js dependencies which is causing failures
in https://github.com/noir-lang/noir/pull/8506

This was broken in
https://github.com/AztecProtocol/aztec-packages/pull/13590 where the
existing msgpack dependency was removed.

Co-authored-by: AztecBot <tech@aztecprotocol.com>

alexghr committed:
feat: store deployment block number for contracts (#14325)

mralj committed:
Merge branch 'feat/enr-client-version' of https://github.com/mralj/aztec-packages into feat/enr-client-version

benesjan committed:
test: fixing flakey `gas_estimation.test.ts` (#14324)

This test was sometimes flakey - an estimated tx fee was different from
the one included in a tx. This happened because we used a stale gas
price to compute the estimated tx fee. I confirmed this by checking the
gas price in the block in which the tx landed with the gas price used
for estimation. It was incorrect to assume that the gas price would stay
constant.

I run the 10 times and confirmed the flake went away (before I got it to
fail like every 3rd time).

While only 1 test case was flakey I decided to do the change in the
other 2 as well to ensure this does not bite us in the future again.

ludamad committed:
chore: increase bb alert thresholds

ludamad committed:
fix: propagate full version string in CLI version prints (#14328)

Closes https://github.com/AztecProtocol/aztec-packages/issues/13949

alexghr committed:
fix: yarn.lock (#14331)

ludamad committed:
fix: revert thread-unsafe bincode parsing (#14336)

This would have broken browser proving

charlielye committed:
chore: yolo noir-contracts debugging.

charlielye committed:
fix: yolo cache upload fix

charlielye committed:
chore: Rework CI flows and benchmarks. (#14202)

Please read [contributing guidelines](CONTRIBUTING.md) and remove this
line.

ludamad committed:
chore(bb): non-avm presets (#14347)

More convenient than setting it manually, saves minutes

Commit:
git subrepo push --branch=master barretenberg

subrepo:
  subdir:   "barretenberg"
  merged:   "b9f61824da"
upstream:
  origin:   "https://github.com/AztecProtocol/barretenberg"
  branch:   "master"
  commit:   "b9f61824da"
git-subrepo:
  version:  "0.4.6"
  origin:   "???"
  commit:   "???" [skip ci]

Commit:
git_subrepo.sh: Fix parent in .gitrepo file.

Commit:
chore: replace relative paths to noir-protocol-circuits

Commit:
git subrepo push --branch=master noir-projects/aztec-nr

subrepo:
  subdir:   "noir-projects/aztec-nr"
  merged:   "5cafc7b98d"
upstream:
  origin:   "https://github.com/AztecProtocol/aztec-nr"
  branch:   "master"
  commit:   "5cafc7b98d"
git-subrepo:
  version:  "0.4.6"
  origin:   "???"
  commit:   "???"

mralj committed:
Merge branch 'feat/enr-client-version' of https://github.com/mralj/aztec-packages into feat/enr-client-version

mralj committed:
import getPackageVersion from update-checker

mralj committed:
getPackageVersion bugfix
